docs/AuthConnectAttachTypes.md

```diff
@@ -0,0 +1,132 @@
+# AUTH_CONNECT Attach Types
+
+## Overview
+
+This document explains the purpose and rationale behind the `BPF_CGROUP_INET4_AUTH_CONNECT` and `BPF_CGROUP_INET6_AUTH_CONNECT` attach types that were added to eBPF for Windows.
+
+## Background
+
+eBPF for Windows already supports several attach types for socket address operations:
+
+- `BPF_CGROUP_INET4_CONNECT` / `BPF_CGROUP_INET6_CONNECT` - Invoked at the redirect layer
+- `BPF_CGROUP_INET4_RECV_ACCEPT` / `BPF_CGROUP_INET6_RECV_ACCEPT` - For incoming connections
+
+However, there was a gap in functionality that required the addition of new attach points.
+
+## Problem Statement
+
+The existing `BPF_CGROUP_INET4_CONNECT` and `BPF_CGROUP_INET6_CONNECT` attach types operate at the **redirect layer** in the Windows Filtering Platform (WFP). This layer occurs **before route selection** in the network stack processing pipeline.
+
+### Limitations of the Redirect Layer
+
+At the redirect layer, certain critical connection properties are not yet available, including:
+
+- **Interface information** - Which network interface will be used for the connection
+- **Tunnel type** - Whether the connection will use tunneling protocols
+- **Route-dependent metadata** - Information that depends on the selected route
+
+### Use Cases Requiring Route Information
+
+eBPF programs that need to make authorization decisions based on these properties cannot function properly with the existing CONNECT attach types. Examples include:
+
+1. **Interface-based policies** - Allowing or denying connections based on which network interface will be used
+2. **Tunnel-aware security** - Different policies for tunneled vs. non-tunneled traffic
+3. **Route-dependent access control** - Authorization decisions that depend on the network path
+
+## Solution: AUTH_CONNECT Attach Types
+
+The new `BPF_CGROUP_INET4_AUTH_CONNECT` and `BPF_CGROUP_INET6_AUTH_CONNECT` attach types address this limitation by operating at the **authorization layer** in WFP.
+
+### Key Characteristics
+
+- **Timing**: Invoked **after route selection** but **before connection authorization**
+- **Available Information**: Full connection context including interface and tunnel information
+- **Purpose**: Authorization decisions based on complete connection metadata
+
+### WFP Layer Mapping
+
+- `BPF_CGROUP_INET4_AUTH_CONNECT` → `FWPM_LAYER_ALE_AUTH_CONNECT_V4`
+- `BPF_CGROUP_INET6_AUTH_CONNECT` → `FWPM_LAYER_ALE_AUTH_CONNECT_V6`
+
+## When to Use Each Attach Type
+
+### Use CONNECT Attach Types When:
+- You need to **redirect** connections to different destinations
+- You only need basic connection information (source/destination IP/port)
```

Comment:
is source information known at this point?

Reply:
I believe so.